src/types/UiaTextRangeBase.cpp

```diff
@@ -339,40 +302,36 @@ IFACEMETHODIMP UiaTextRangeBase::ExpandToEnclosingUnit(_In_ TextUnit unit)
 
     try
     {
-        const ScreenInfoRow topRow = _getFirstScreenInfoRowIndex();
-        const ScreenInfoRow bottomRow = _getLastScreenInfoRowIndex(_pData);
+        const auto& buffer = _pData->GetTextBuffer();
+        const auto bufferSize = buffer.GetSize();
+        const auto bufferEnd = bufferSize.EndExclusive();
 
         if (unit == TextUnit::TextUnit_Character)
         {
             _end = _start;
+            bufferSize.IncrementInBounds(_end, true);
         }
         else if (unit <= TextUnit::TextUnit_Word)
         {
             // expand to word
-            const auto target = _start;
-            _start = _wordBeginEndpoint(_pData, target, _wordDelimiters);
-            _end = _wordEndEndpoint(_pData, target, _wordDelimiters);
-            FAIL_FAST_IF(!(_start <= _end));
+            _start = buffer.GetWordStart(_start, _wordDelimiters, /*accessibilityMode*/ true);
+            _end = buffer.GetWordEnd(_start, _wordDelimiters, /*accessibilityMode*/ true);
         }
         else if (unit <= TextUnit::TextUnit_Line)
         {
             // expand to line
-            _start = _textBufferRowToEndpoint(_pData, _endpointToTextBufferRow(_pData, _start));
-            _end = _start + _getLastColumnIndex(_pData);
-            FAIL_FAST_IF(!(_start <= _end));
+            _start.X = 0;
+            _end.X = base::ClampAdd(_start.Y, 1);

```

Comment:
wait why is this `x=y+1`? Should these both be `y` (or `x`)?

Reply:
oof. Yeah. I also got rid of `_end.X=0`. That was a big nono. Good catch!